icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/BurmeseBreakEngine.java

```diff
@@ -31,23 +31,22 @@
     private static final byte BURMESE_MIN_WORD = 2;
 
     private DictionaryMatcher fDictionary;
-    private static UnicodeSet fBurmeseWordSet;
-    private static UnicodeSet fEndWordSet;
-    private static UnicodeSet fBeginWordSet;
-    private static UnicodeSet fMarkSet;
+    private UnicodeSet fEndWordSet;
```

Comment:
Note: The trade-off here is to permanently hold onto less memory, but to increase the time it takes to construct a break engine after the first time. I guess it's ok either way until someone complains about memory or about performance.

Reply:
as I mention, for all these LanguageBreakEngine, since they are not part of the ICU API, they are only used internally in Java.
The only place construct them inside Java is inside
main/classes/core/src/com/ibm/icu/text/RuleBasedBreakIterator.java 
and tracked by     private static final List<LanguageBreakEngine> gAllBreakEngines;
and all LanguageBreakEngine (except CjkBreakEngine) will only be created once and remember in gAllBreakEngines and will not call the constructor more than once. CjkBreakEngine constructor may be called twice- once for UScript.HAN UScript.KATAKANA or  UScript.HIRAGANA (shared by one) and another time for UScript.HANGUL . Therfore the so-called "construct a break engine after the first time" will never happen except the second time for CjkBreakEngine .

Reply:
Sounds good. Thanks for the detailed explanation, all makes sense. I just didn't have the facts right.